compiler/optimizer/Operation.cpp

```diff
@@ -0,0 +1,347 @@
+         break;
+      }
+      case TR::lrol:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         operand2 = nodeValueMap[node->getChild(1)->getGlobalIndex()];
+         result.type = Int64;
+         result.data.lconst = (operand1.data.lconst << operand2.data.lconst) 
+                              | (operand1.data.lconst >> (64 - operand2.data.lconst));
+         break;
+      }
+      case TR::land:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         operand2 = nodeValueMap[node->getChild(1)->getGlobalIndex()];
+         result.type = Int64;
+         result.data.lconst = operand1.data.lconst & operand2.data.lconst;
+         break;
+      }
+      case TR::lor:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         operand2 = nodeValueMap[node->getChild(1)->getGlobalIndex()];
+         result.type = Int64;
+         result.data.lconst = operand1.data.lconst | operand2.data.lconst;
+         break;
+      }
+      case TR::lxor:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         operand2 = nodeValueMap[node->getChild(1)->getGlobalIndex()];
+         result.type = Int64;
+         result.data.lconst = operand1.data.lconst ^ operand2.data.lconst;
+         break;
+      }
+      case TR::l2i:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         result.type = Int32;
+         result.data.iconst = (int32_t)operand1.data.lconst;
+         break;
+      }
+      case TR::l2f:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         result.type = Float;
+         result.data.fconst = (float_t)operand1.data.lconst;
+         break;
+      }
+      case TR::l2d:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         result.type = Double;
+         result.data.dconst = (double_t)operand1.data.lconst;
+         break;
+      }
+      case TR::l2b:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         result.type = Int8;
+         result.data.bconst = (int8_t)operand1.data.lconst;
+         break;
+      }
+      case TR::l2s:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         result.type = Int16;
+         result.data.sconst = (int16_t)operand1.data.lconst;
+         break;
+      }
+      case TR::l2a:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         result.type = Address;
+         result.data.aconst = (uintptrj_t)operand1.data.lconst;
+         break;
+      }
+      case TR::lcmpeq:
+      {
+         operand1 = nodeValueMap[node->getChild(0)->getGlobalIndex()];
+         operand2 = nodeValueMap[node->getChild(1)->getGlobalIndex()];
+         result.type = Boolean;
+         result.data.lconst = operand1.data.lconst == operand2.data.lconst ? true : false;
```

Comment:
@andrewcraik suggested we store booleans in `lconst`. We changed result type to `Int64` and assigned `1` and `0` instead of `true` and `false` (here and elsewhere). What do you think?

Reply:
In the IL, there is no explicit boolean type. So, using an `Int64` is more consistent with how the IL works (technically, the IL uses Int32, not Int64, for boolean values but that's a detail we can worry about later when other Int32 operations are supported). What matters is that the type tag and the field of the `.data` union assigned to should match. Otherwise, you're missing the value of having the type tag in the first place.